src/main/java/lotto/domain/MatchCriteria.java

```diff
@@ -0,0 +1,52 @@
+package lotto.domain;
+
+public enum MatchCriteria {
+    THREE(new Criteria(3), new Amount(5_000)),
+    FOUR(new Criteria(4), new Amount(50_000)),
+    FIVE(new Criteria(5), new Amount(1_500_000)),
+    SIX(new Criteria(6), new Amount(2_000_000_000));
+
+    private final static int INIT_COUNT = 0;
+    private final Criteria criteria;
+    private final Amount prize;
+    private Count count;
+
+    MatchCriteria(Criteria criteria, Amount prize) {
+        this.criteria = criteria;
+        this.prize = prize;
+        initCount();
+    }
+
+    public void initCount() {
+        this.count = new Count(INIT_COUNT);
+    }
+
+    public Amount calc() {
+        return prize.multiflyCountAmount(count);
+    }
+
+    public MatchCriteria match() {
+        count = count.increasedCount();
+        return this;
+    }
```

Comment:
반환되는 MatchCriteria를 활용하는 곳이 없네요. void로 변경해도 좋을 것 같아요.
```suggestion
    public void match() {
        count = count.increasedCount();
    }
```